app/src/main/java/com/example/bigowlapp/utils/IsSmsSenderACurrentEventSupervisor.java

```diff
@@ -0,0 +1,69 @@
+package com.example.bigowlapp.utils;
+
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.List;
+
+public class IsSmsSenderACurrentEventSupervisor extends AppCompatActivity {
+
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+    public IsSmsSenderACurrentEventSupervisor() {
+
+    }
+
+    public Task<List<Schedule>> check(String phoneNumber) {
+        RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+        String currentUserUid = repositoryFacade.getAuthRepository().getCurrentUser().getUid();
+
+        Task<QuerySnapshot> gettingUserTask = db.collection("users")
+                .whereEqualTo("phoneNumber", phoneNumber)
+                .get();
+        Task<User> userTask = gettingUserTask.continueWithTask(task -> {
+            if (task.isSuccessful()) {
+                if (!task.getResult().isEmpty()) {
+                    List<User> listOfUsers = task.getResult().toObjects(User.class);
+                    return Tasks.forResult(listOfUsers.get(0));
+                } else throw task.getException();
+            } else throw task.getException();
+        });
+
+        Task<List<Schedule>> schedulesTask = userTask.continueWithTask(task -> {
```

Comment:
Inline variable. you could just return the assignment value. Same goes to line 48

Reply:
Having it  like this just helped to understand and read the code easily